core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -227,13 +215,43 @@ public HttpServiceModule getHttpLayer() {
     }
 
     /**
-     * Returns the configuration for this singleThreadService from the JSON config file
+     * Returns the configurations for this service type from the JSON config file.
+     *
+     * @param type Service type
+     * @return the configurations for this service type from the JSON config file
+     */
+    public List<ServiceConfiguration> getServiceConfigurationsByType(final String type) {
+        return serviceConfigsByType.get(type);
+    }
+
+    /**
+     * Returns the configuration for this service type from the JSON config file.
+     *
+     * If there are multiple configs for the type, the first one will be returned.
+     *
+     * @param type Service type
+     * @return the first configuration for this service type from the JSON config file
+     */
+    public ServiceConfiguration getServiceConfigurationByType(final String type) {
```

Comment:
I think we should not have this one. By type should always return a list.

Reply:
@danielpassos how about renaming it to "getSingleServiceConfigByType" ?

otherwise, I will have to duplicate the content of that method in the callers.

Reply:
@danielpassos Also, John recommends having a single configs for services: 
So, I would like to keep that method.